gef-ui/server/routes/application-details.js

```diff
@@ -1,10 +1,11 @@
 const express = require('express');
 const { applicationDetails, postApplicationDetails } = require('../controllers/application-details');
-const { validateToken, validateBank } = require('../middleware');
+const { validateToken, validateBank, validateRole } = require('../middleware');
+const { MAKER, CHECKER, READ_ONLY } = require('../constants/roles');
 
 const router = express.Router();
 
-router.get('/application-details/:dealId', [validateToken, validateBank], (req, res) => applicationDetails(req, res));
-router.post('/application-details/:dealId', [validateToken, validateBank], (req, res) => postApplicationDetails(req, res));
+router.get('/application-details/:dealId', [validateToken, validateRole({ role: [MAKER, CHECKER, READ_ONLY] }), validateBank], (req, res) => applicationDetails(req, res));
```

Comment:
What about `ADMIN`?

Reply:
@abhi-markan requested 
`application-details/:dealId - Maker, checker, read-only please.`

Reply:
Okay, I think we should double-check as it was my understanding that `admin` has the same rights as `read-only` plus being able to edit users

Reply:
updated after checking